### PR TITLE
Mitchell/ad 224 sql function support

### PR DESCRIPTION
**Pull request by mitchell-elholm**
### Summary

Add Tests for SQL Functions

### Description

Added tests in WHERE clause for:

- Division
- Multiplication
- Modulo
- Addition
- Subtraction
- Boolean operators (AND OR NOT)
- Date addition
- Date subtraction
- MONTHNAME
- DAYNAME
- EXTRACT
- CURRENT_DATE

### Related Issue

https://bitquill.atlassian.net/browse/AD-224

### Additional Reviewers
@birschick-bq
@mitchell-elholm
@alexey-temnikov 
@andiem-bq

### CODE REVIEW

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementFilterTest.java [hunk 1]
```diff
@@ -1026,4 +1026,416 @@ void testFloorForDateInWhere() throws SQLException {
...
+        Assertions.assertTrue(resultSet.next());
+        Assertions.assertEquals("101", resultSet.getString(1));
+        Assertions.assertTrue(resultSet.next());
+        Assertions.assertEquals("102", resultSet.getString(1));
+        Assertions.assertFalse(resultSet.next());
+    }
+
+    @Test
+    @DisplayName("Tests queries filtering by date extract.")
+    void testWhereExtract() throws SQLException {
```
Comment (andiemontoyeah) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementFilterTest.java:1289]: Let's delete the older one in line 454 testQueryWhereExtract?

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java [hunk 2]
```diff
@@ -480,6 +485,7 @@ void testFloorForDate() throws SQLException {
                         + " [{\"$subtract\": [\"$field\", {\"$date\": \"1970-01-01T00:00:00Z\"}]},"
                         + " {\"$mod\": [{\"$subtract\": [\"$field\", {\"$date\": \"1970-01-01T00:00:00Z\"}]},"
                         + " 1]}]}, 1]}]}]}}}").toJson(),
+
```
Comment (alexey-temnikov) [inline calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java]: nit: extra empty line

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementFilterTest.java (hunk 1 above)
Comment (mitchell-elholm) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementFilterTest.java:1289]: Removed duplicate test.

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java (hunk 2 above)
Comment (mitchell-elholm) [reply calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java]: Removed extra line.

**Review by andiemontoyeah — APPROVED** [review]
lgtm!

**Pull request merged**